pkg/controller/machine/machine_controller.go

```diff
@@ -114,8 +114,10 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	// Cluster might be nil as some providers might not require a cluster object
 	// for machine management.
-	cluster, err := r.getCluster(ctx, m)
-	if err != nil {
+	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, m.ObjectMeta)
+	if err == util.ErrNoCluster {
+		klog.Infof("Machine %q in namespace %q doesn't specify %q label, assuming nil cluster", m.Name, m.Namespace, clusterv1.MachineClusterLabelName)
+	} else if err != nil {
```

Comment:
I prefer to avoid `else`s like this:
```go
if err == util.ErrNoCluster {
  // ...
}
if err != nil {
  // ...
}
```

Reply:
I don't really like how the blocks reads in that case, I can change it if you have strong feelings about it

Reply:
It's been my coding style for the past few years.... oh but wait, you're not returning early from the ErrNoCluster case. Nevermind.